pkg/logging/logging.go

```diff
@@ -1,140 +1,115 @@
-	switch strings.ToLower(encoding) {
-	case "json", "console", "logfmt":
-	default:
-		return fmt.Errorf("unknown encoding %q", encoding)
-	}
-
-	var level zap.AtomicLevel
-	switch strings.ToLower(logLevel) {
-	case "debug":
-		level = zap.NewAtomicLevelAt(zapcore.DebugLevel)
-	case "info":
-		level = zap.NewAtomicLevelAt(zapcore.InfoLevel)
-	case "warn":
-		level = zap.NewAtomicLevelAt(zapcore.WarnLevel)
-	case "error":
-		level = zap.NewAtomicLevelAt(zapcore.ErrorLevel)
-	case "fatal":
-		level = zap.NewAtomicLevelAt(zapcore.FatalLevel)
-	case "panic":
-		level = zap.NewAtomicLevelAt(zapcore.PanicLevel)
-	default:
-		return fmt.Errorf("unsupported log level %q", logLevel)
-	}
-
-	zapEncoderConfig := zapcore.EncoderConfig{
-		TimeKey:        "ts",
-		LevelKey:       "level",
-		NameKey:        "logger",
-		CallerKey:      "caller",
-		MessageKey:     "msg",
-		StacktraceKey:  "stacktrace",
-		LineEnding:     zapcore.DefaultLineEnding,
-		EncodeLevel:    zapcore.LowercaseLevelEncoder,
-		EncodeTime:     zapcore.ISO8601TimeEncoder,
-		EncodeDuration: zapcore.MillisDurationEncoder,
-		EncodeCaller:   zapcore.ShortCallerEncoder,
-	}
-
-	// registering the encoder is taken care of the by teh zaplogfmt library itself
-	if encoding == "logfmt" {
-		zapEncoderConfig.EncodeTime = func(ts time.Time, encoder zapcore.PrimitiveArrayEncoder) {
-			encoder.AppendString(ts.UTC().Format(time.RFC3339))
+func Init(version, logLevel, encoding string, opts ...Option) error {
+	var level = new(slog.Level)
+	err := level.UnmarshalText([]byte(logLevel))
+	if err != nil {
+		return fmt.Errorf("unsupported log level %q: %w", logLevel, err)
+	}
+
+	replaceFunc := func(groups []string, a slog.Attr) slog.Attr {
+		// write time as ts
+		switch a.Key {
+		case slog.TimeKey:
+			a.Key = "ts"
+		case slog.LevelKey:
+			// Handle custom level values
+			level := a.Value.Any().(slog.Level)
+
+			switch {
+			case level < LevelInfo:
+				a.Value = slog.StringValue(debugLevel)
+			case level < LevelWarn:
+				a.Value = slog.StringValue(infoLevel)
+			case level < LevelError:
+				a.Value = slog.StringValue(warnLevel)
+			case level < LevelFatal:
+				a.Value = slog.StringValue(errorLevel)
+			case level < LevelPanic:
+				a.Value = slog.StringValue(fatalLevel)
+			default:
+				a.Value = slog.StringValue(panicLevel)
+			}
+		case slog.SourceKey:
+			a.Key = "caller"
+
+			// only returns the pkg name, file and line number
+			dir, file := filepath.Split(a.Value.String())
+			a.Value = slog.StringValue(filepath.Join(filepath.Base(dir), file))
 		}
+		return a
 	}
 
-	cfg := zap.Config{
-		Level: level,
-		InitialFields: map[string]interface{}{
-			"app_name":    appName,
-			"app_version": appVersion,
-		},
-		Encoding:         encoding,
-		EncoderConfig:    zapEncoderConfig,
-		OutputPaths:      []string{"stdout"},
-		ErrorOutputPaths: []string{"stderr"},
+	cfg := &loggingConfig{
+		enableCaller: true,
+		output:       os.Stdout,
```

Comment:
Out of curiosity: I've seen loggers handle stuff below a certain level on e.g. `STDOUT`, while anything above is emitted to `STDERR`. Is that something we want / need to support?

Reply:
Something to think about when we move the package into its own. Would require a handler implementation that splits calls to a stdout (or normal) and to a stderr (or error) handler.